test/functional/cypress/integration/ecs/server_group.spec.js

```diff
@@ -163,19 +165,50 @@ describe('amazon ecs: ECSApp Server Group', () => {
     cy.get('[data-test-id="ServerGroup.details"]').clear().type('computeOptions');
+    cy.get('[data-test-id="ServerGroup.clusterName"]').type('example-app-test-Cluster-NSnYsTXmCfV2');
+    cy.get('span:contains("example-app-test-Cluster-NSnYsTXmCfV2")').click();
 
     cy.get('[data-test-id="ServerGroup.computeOptionsCapacityProviders"]').click();
+    cy.get('[data-test-id="ServerGroup.capacityProviders.default"]').click();
+
+    cy.get('[data-test-id="ServerGroupWizard.submitButton"]').click();
+    cy.get('.glyphicon-edit').click();
+
+    cy.get('[data-test-id="ServerGroup.defaultCapacityProvider.name.0"]').should('have.value', 'FARGATE_SPOT');
+    cy.get('[data-test-id="ServerGroup.capacityProvider.base.0"]').should('have.value', '0');
+    cy.get('[data-test-id="ServerGroup.capacityProvider.weight.0"]').should('have.value', '1');
+
+    cy.get('[data-test-id="ServerGroupWizard.submitButton"]').click();
+    cy.get('[data-test-id="Pipeline.revertChanges"]').click();
+  });
+
+  it('edit an existing server group to use custom capacity providers', () => {
+    cy.visit('#/applications/ecsapp/executions');
+
+    cy.get('a:contains("Configure")').click();
+    cy.get('a:contains("Deploy")').click();
+    cy.get('.glyphicon-edit').click();
+
+    cy.get('[data-test-id="ServerGroup.stack"]').clear().type('edit');
+    cy.get('[data-test-id="ServerGroup.details"]').clear().type('computeOptions');
+    cy.get('[data-test-id="ServerGroup.clusterName"]').type('example-app-test-Cluster-NSnYsTXmCfV2');
+    cy.get('span:contains("example-app-test-Cluster-NSnYsTXmCfV2")').click();
+
+    cy.get('[data-test-id="ServerGroup.computeOptionsCapacityProviders"]').click();
+    cy.get('[data-test-id="ServerGroup.capacityProviders.custom"]').click();
     cy.get('[data-test-id="ServerGroup.addCapacityProvider"]').click();
-    cy.get('[data-test-id="capacityProvider.name.0"]').type('FARGATE');
-    cy.get('[data-test-id="capacityProvider.base.0"]').type(1);
-    cy.get('[data-test-id="capacityProvider.weight.0"]').type(2);
+
+    cy.get('[data-test-id="ServerGroup.customCapacityProvider.name.0"]').type('FARGATE_SPOT');
+    cy.get('.Select-option:contains("FARGATE_SPOT")').click();
+    cy.get('[data-test-id="ServerGroup.capacityProvider.base.0"]').type(1);
+    cy.get('[data-test-id="ServerGroup.capacityProvider.weight.0"]').type(2);
 
     cy.get('[data-test-id="ServerGroupWizard.submitButton"]').click();
     cy.get('.glyphicon-edit').click();
 
-    cy.get('[data-test-id="capacityProvider.name.0"]').should('have.value', 'FARGATE');
-    cy.get('[data-test-id="capacityProvider.base.0"]').should('have.value', '1');
-    cy.get('[data-test-id="capacityProvider.weight.0"]').should('have.value', '2');
+    cy.get('[data-test-id="ServerGroup.customCapacityProvider.name.0"]').children().children().children(".Select-multi-value-wrapper").children().children('.Select-value-label').should('contain', 'FARGATE_SPOT');
```

Comment:
this is a lot of `.children()` - can we not add a test ID to the element we want to access?

Reply:
I am completely in favor of adding test ID to the element than using so many children(), but unfortunately we are using TetheredSelect component from @spinnaker/core where we can not assign the ID to the select element directly. That is why I had to use so many children to reach to the required element.

Reply:
You don't need to use.children.childern (take a look at how we do testing for container images, and other tethered select components: https://github.com/spinnaker/deck/blob/master/test/functional/cypress/integration/ecs/server_group.spec.js#L108)